core/src/main/java/org/web3j/protocol/core/methods/response/AbiDefinition.java

```diff
@@ -275,12 +275,11 @@ public void setComponents(final List<NamedType> components) {
             this.components = components;
         }
 
-        public int structIdentifier() {
+        public String structIdentifier() {
             return ((internalType == null ? type : internalType.isEmpty() ? type : internalType)
                             + components.stream()
-                                    .map(namedType -> String.valueOf(namedType.structIdentifier()))
-                                    .collect(Collectors.joining()))
-                    .hashCode();
```

Comment:
hashCode does not have uniqueness Property, it could subject to quite significant collision as a identifier